src/analysis.py

```diff
@@ -0,0 +1,219 @@
+        sys.exit()
+
+    print('Returning data frame')
+    return data_frame
+
+#This function is adapted from Varada's lecture code in DSCI571
+def store_cross_val_results(model, scores, results):
+    """
+    Stores mean scores from cross_validate in results_dict for
+    the given model model_name.
+
+    Parameters
+    ----------
+    model :
+        scikit-learn classification model
+    scores : dict
+        object return by `cross_validate`
+    results_dict: dict
+        dictionary to store results
+
+    Returns
+    ----------
+        None
+
+    """
+    results[model] = {
+        "mean_validation_accuracy": "{:0.4f}".format(np.mean(scores["test_score"])),
+        "mean_train_accuracy": "{:0.4f}".format(np.mean(scores["train_score"])),
+        "mean_fit_time (s)": "{:0.4f}".format(np.mean(scores["fit_time"])),
+        "mean_score_time (s)": "{:0.4f}".format(np.mean(scores["score_time"])),
+        "std_train_score": "{:0.4f}".format(scores["train_score"].std()),
+        "std_test_score": "{:0.4f}".format(scores["test_score"].std()),
+    }
+
+
+
+def train_model(dataframe):
+    """
+    Processes the data, trains the model, and returns a dataframe showing the results
+
+    Parameters
+    ----------
+    dataframe : pd.Dataframe
+        Dataframe to be used in model
+
+    Returns
+    ----------
+        pandas.Dataframe
+
+    """
+    train_df = dataframe
+    X_train = train_df.drop(columns=['align'])
+    y_train = train_df['align']
+
+    numeric_features = ['appearances']
+    categorical_features = ['name', 'id', 'eye', 'hair', 'year', 'publisher', 'sex']
+    drop_features = ['gsm', 'first_appearance']
+
+    numeric_transformer = Pipeline(
+        steps=[
+            ("imputer", SimpleImputer(strategy='median')),
+            ("scaler", StandardScaler()),
+        ]
+    )
+
+    categorical_transformer = Pipeline(
+        steps=[
+            ("imputer", SimpleImputer(strategy='most_frequent')),
+            ("onehot", OneHotEncoder(handle_unknown='ignore'))
+        ]
+    )
+
+    preprocessor = make_column_transformer(
+        ("drop", drop_features),
+        (numeric_transformer, numeric_features),
+        (categorical_transformer, categorical_features),
+    )
+
+    results_df = {}
+
+    models = {
+        "Dummy Classifier": DummyClassifier(),
+        "Decision Tree Classifier": DecisionTreeClassifier(),
+        "Random Forest Classifier": RandomForestClassifier(),
+        "SVM": SVC(),
+        "OVR LogisticRegression": OneVsRestClassifier(LogisticRegression()),
+        "LogisticRegression": LogisticRegression(),
+    }
+
+    for key, value in models.items():
+        pipeline = Pipeline(
+                steps=[("preprocessor", preprocessor),
+               (key, value)
+                ]
+            )
+        scores = cross_validate(pipeline, X_train, y_train, return_train_score=True, n_jobs=-1, verbose=1)
```

Comment:
Do we need to run on the test set?

Reply:
I am running out of time relative to other labs I have to get done... As discussed per milestones thing this would be a nice to have for this week but not must have (given we have attempted hyperparameter optimization)

Reply:
Okay, if I have time later I may try to throw it in. We'll just see how it goes for now.